File: app/src/main/java/com/cradletrial/cradlevhtapp/view/SplashActivity.java
package com.cradletrial.cradlevhtapp.view;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.v7.app.AppCompatActivity;

import com.cradletrial.cradlevhtapp.R;

public class SplashActivity extends AppCompatActivity {

    private final int DELAY = 100;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                Intent intent = new Intent(SplashActivity.this, LoginActivity.class);
                startActivity(intent);
                finish();
            }
        }, DELAY);
    }
}

File: app/src/main/java/com/cradle/neptune/view/SplashActivity.kt
package com.cradle.neptune.view

import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import androidx.core.os.postDelayed
import androidx.preference.PreferenceManager
import com.cradle.neptune.R
import com.cradle.neptune.manager.LoginManager
import com.cradle.neptune.utilitiles.SharedPreferencesMigration
import dagger.hilt.android.AndroidEntryPoint
import kotlinx.coroutines.runBlocking
import javax.inject.Inject

@AndroidEntryPoint
class SplashActivity : AppCompatActivity() {

    @Inject
    lateinit var loginManager: LoginManager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)

        val sharedPreferences = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        val isMigrationSuccessful = SharedPreferencesMigration(sharedPreferences).migrate()
        if (!isMigrationSuccessful) {
            runBlocking {
                loginManager.logout()
            }
        }

        Handler().postDelayed(DELAY_MILLIS) {
            val intent = if (isMigrationSuccessful) {
                LoginActivity.makeIntent(this, shouldDisplayMessage = false)
            } else {
                LoginActivity.makeIntent(
                    this,
                    shouldDisplayMessage = true,
                    displayMessageTitleRes = R.string.logout_due_to_error_dialog_title,
                    displayMessageBodyRes = R.string.logout_due_to_error_dialog_message
                )
            }
            startActivity(intent)
            finish()
        }
    }

    companion object {
        private const val DELAY_MILLIS = 100L
    }
}
